ledger/participant-integration-api/src/test/lib/scala/platform/indexer/ha/IndexerStabilityTestFixture.scala

```diff
@@ -27,7 +29,7 @@ case class ReadServiceAndIndexer(
 case class Indexers(indexers: List[ReadServiceAndIndexer]) {
   // The list of all indexers that are running (determined by whether they have subscribed to the read service)
   def runningIndexers: List[ReadServiceAndIndexer] =
-    indexers.filter(x => x.readService.stateUpdatesCalls.get() > 0 && !x.readService.aborted)
+    indexers.filter(_.readService.isRunning)
```

Comment:
I think this is a nice improvement. I also think there was a crack in the concurrency approach with the previous solution as aborted was used non-synchronized (visibility issues), and stateUpdateCalls and aborted used non semaphored after each other.

An idea for further improvements: having the ReadService itself separated from the abort/reset logic might be beneficial, not sure it is worth the effort though.

And my usual chant: can it be that improving EndlessReadService's concurrency model is well placed in it's own PR?

Reply:
I agree with having abort/reset logic separated - this is what I meant by being `not happy` with -  `IndexerStabilityTestFixture` and `IndexerStabilitySpec` and the way it integrates with `ReadService`.  And it deserves it's own PR.